json_object_private.h

```diff
@@ -43,36 +45,53 @@ struct json_object
 	json_object_private_delete_fn *_delete;
 	json_object_to_json_string_fn *_to_json_string;
```

Comment:
Is it possible to extract these functions into public interface for further reduce the size of data structure? Three func pointers will occupy 24 bytes in a 64-bit system. This is also very effective.

Reply:
Are you suggesting, e.g., changing json_object_to_json_string_ext() to take a json_object_to_json_string_fn argument?  That won't work for two reason:
1. it changes the public API, and I definitely don't want to do that as part of _this_ change.
2. More importantly, it won't work, because each json_object is allowed to have a _different_ to_json_string function, specified by callers of the json-c API.

On the other hand, _delete can't be changed by external callers, so *that* we could potentially turn into direct calls in json_object_put.  Good idea, I'll take a stab at doing that.

Reply:
See commit 5ebfeaed